astropy/io/fits/card.py

```diff
@@ -66,7 +66,7 @@ class Card(_Verify):
     # followed by an optional comment
     _strg = r"\'(?P<strg>([ -~]+?|\'\'|) *?)\'(?=$|/| )"
     _comm_field = r"(?P<comm_field>(?P<sepr>/ *)(?P<comm>(.|\n)*))"
-    _strg_comment_RE = re.compile(f"({_strg})? *{_comm_field}?")
+    _strg_comment_RE = re.compile(f"({_strg})? *{_comm_field}?$")
```

Comment:
Second issue, a string after the double quotes is lost if there is a space in between:
```
In [7]: fits.Card.fromstring(fits.Card("FOO", "x"*100 + "'' aaa", "comment").image).value
Out[7]: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'"
```

`_value_NFSC_RE` doesn't do that for normal cards, and the difference is the `$` at the end of the regex. Not sure why, but to confirm this:
https://regex101.com/r/kuMQfG/1
https://regex101.com/r/kuMQfG/2